Application/EdFi.Ods.AdminApi/Infrastructure/Services/ClaimSetEditor/GetResourcesByClaimSetIdQueryV6Service.cs

```diff
@@ -156,34 +156,36 @@ void AddStrategyToParentResource(IEnumerable<SecurityAuthorizationStrategy>? def
                                        x.ResourceClaim.ResourceClaimId == resourceClaim.ParentResourceClaimId &&
                                        x.Action.ActionName == action.Value)?.AuthorizationStrategies.Select(x => x.AuthorizationStrategy);
 
-                            // SingleOrDefault means we could get a null back,
-                            // and the next line is written to assume we'll
-                            // never get a null. This needs to be refactored to
-                            // be more defensive.
-                            var mappedStrategies = defaultStrategies!.Select(x =>
+                            if (defaultStrategies != null)
                             {
-                                var value = _mapper.Map<AuthorizationStrategy>(x);
-                                if (value != null)
-                                    value.IsInheritedFromParent = true;
-                                return value;
-                            });
-
-                            actions.Add(new ClaimSetResourceClaimActionAuthStrategies()
-                            {
-                                AuthorizationStrategies = mappedStrategies.ToArray()
-                            });
+                                var mappedStrategies = defaultStrategies.Select(x =>
+                                {
+                                    var value = _mapper.Map<AuthorizationStrategy>(x);
+                                    if (value != null)
+                                        value.IsInheritedFromParent = true;
+                                    return value;
+                                });
+
+                                actions.Add(new ClaimSetResourceClaimActionAuthStrategies()
+                                {
+                                    AuthorizationStrategies = mappedStrategies.ToArray()
+                                });
+                            }
                         }
                         else
                         {
-                            var mappedStrategies = defaultStrategies!.Select(x =>
+                            if (defaultStrategies != null)

```

Comment:
This check is unnecessary because of line 153 above.

Reply:
We are rehydrating the "defaultStrategies" in line 155. It is "SingleOrDefault", so it can be null